internal/awslambda/extension/manager.go

```diff
@@ -214,6 +230,41 @@ func (m *manager) register(ctx context.Context) error {
 	return nil
 }
 
+func (m *manager) subscribeToTelemetry(ctx context.Context) error {
+	b, err := jsoniter.Marshal(telemetry.SubscriptionRequest{
+		SchemaVersion: telemetry.ApiVersion,
+		Types:         []string{telemetry.PlatformSubscriptionType},
+		Destination:   &telemetry.SubscriptionDestination{URI: m.telemetryServer.Endpoint(), Protocol: telemetry.ProtocolHTTP},
+		// we buffer for the minimum time on the lambda side to avoid adding additional billedDuration to the lambda
+		Buffering: &telemetry.SubscriptionBufferingConfig{TimeoutMs: telemetry.MinBufferingTimeoutMs},
+	})
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, telemetry.SubscribeEndpoint.GetUrl(m.domain), bytes.NewReader(b))
```

Comment:
```suggestion
	req, err := http.NewRequestWithContext(
		ctx,
		http.MethodPut,
		telemetry.SubscribeEndpoint.GetUrl(m.domain),
		bytes.NewReader(b),
	)
	if err != nil {
		return err
	}
```
Missing check for the returned error